modules/core/src/main/java/org/apache/ignite/internal/managers/communication/GridIoManager.java

```diff
@@ -1593,6 +1658,75 @@ public boolean checkNodeLeft(UUID nodeId, IgniteCheckedException sndErr, boolean
             (ping && !ctx.discovery().pingNode(nodeId));
     }
 
+    /**
+     * @param node Destination node to connect to.
+     * @param topic Topic to send the request to.
+     * @param initMsg Channel initialization message.
+     * @return Established {@link Channel} to use.
+     * @throws IgniteCheckedException If fails.
+     */
+    private IgniteInternalFuture<Channel> openChannel(
+        ClusterNode node,
+        Object topic,
+        Message initMsg
+    ) throws IgniteCheckedException {
+        assert node != null;
+        assert topic != null;
+
+        if (locNodeId.equals(node.id()))
+            throw new IgniteCheckedException("Channel cannot be opened to the local node itself " +
+                "[nodeId=" + node.id() + ", topic=" + topic + ']');
+
+        int topicOrd = topic instanceof GridTopic ? ((Enum<GridTopic>)topic).ordinal() : -1;
+
+        GridIoMessage ioMsg = createGridIoMessage(topic, topicOrd, initMsg, PUBLIC_POOL, false, 0, false);
+
+        try {
+            if ((CommunicationSpi)getSpi() instanceof TcpCommunicationSpi) {
```

Comment:
It' will be more readable if we invert condition
```
if (! (instanceOf XXX)) {
   throw new ScaryException("XXX");
}

return ALL_OK;
```

Reply:
Agree, fixed.